ext/Geant4/io_gdml.jl

```diff
@@ -348,7 +353,48 @@ function parse_geometry(e::Torus, x_solids::XMLElement, x_define::XMLElement, id
             "lunit" => SolidStateDetectors.internal_length_unit,
             "aunit" => "deg"
         ))
-    
+    end
+end
+
+function _rectangle_point(x::T)::Tuple{T,T} where {T <: Real}
+    return (
+        clamp(tand(abs(mod(x-90,360)-180)-90),-1,1),
+        clamp(tand(abs(mod(   x,360)-180)-90),-1,1)
+    )
+end
```

Comment:
Some back info: For tori with partial `θ` ranges, I intersect a full torus with a `PolyCone`.
The function `_rectangle_point` returns a point on a rectangle at a given `θ` (in degrees):
```julia
using SolidStateDetectors
using Geant4
using Plots

const G4 = Base.get_extension(SolidStateDetectors, :SolidStateDetectorsGeant4Ext)

base_plot = plot([1,1,-1,-1,1],[1,-1,-1,1,1], color = :black, ratio = 1, size = (400,400), label = "", xlims = (-2,2), ylims = (-2,2))
@gif for x in 0:5.:360-5
    p = G4._rectangle_point(x)
    plot(deepcopy(base_plot))
    scatter!([p[1]], [p[2]], label = "", color = :black)
end
```
![tmp](https://github.com/user-attachments/assets/4c1719ca-0bc2-4042-a541-bb0bc47ad2a4)